Gems/ScriptCanvas/Code/Editor/Components/EditorGraph.cpp

```diff
@@ -1130,10 +1135,75 @@ namespace ScriptCanvasEditor
         }
 
-
+        UpdateCorrospondingImplicitConnection(scSourceEndpoint, scTargetEndpoint);
+        
         return scConnected;
     }
 
+    void EditorGraph::UpdateCorrospondingImplicitConnection(const ScriptCanvas::Endpoint& sourceEndpoint, const ScriptCanvas::Endpoint& targetEndpoint)
+    {
+        const ScriptCanvas::Node* sourceNode = FindNode(sourceEndpoint.GetNodeId());
+        const ScriptCanvas::Node* targetNode = FindNode(targetEndpoint.GetNodeId());
+
+        if (!sourceNode || !targetNode)
+        {
+            return;
+        }
+
+        // Retrieve the source node's execution out and target node's execution in
+        const ScriptCanvas::Slot* sourceNodeExecutionSlot;
+        const ScriptCanvas::Slot* targetNodeExecutionSlot;
+
+        AZStd::vector< const ScriptCanvas::Slot* > sourceNodeExecutionSlots = sourceNode->GetAllSlotsByDescriptor(ScriptCanvas::SlotDescriptors::ExecutionOut(), false);
+        AZStd::vector< const ScriptCanvas::Slot* > targetNodeExecutionSlots = targetNode->GetAllSlotsByDescriptor(ScriptCanvas::SlotDescriptors::ExecutionIn(), false);
+
+        // Ensure there is exactly 1 non-latent execution source and target slot
+        if (sourceNodeExecutionSlots.size() == 1 && targetNodeExecutionSlots.size() == 1)
+        {
+            sourceNodeExecutionSlot = sourceNodeExecutionSlots[0];
+            targetNodeExecutionSlot = targetNodeExecutionSlots[0];
+        }
+        else
+        {
+            return;
+        }
+
+        // If either the source or target slot execution slots on these nodes are implicit, then check if the implicit connection should be updated
+        if (sourceNodeExecutionSlot->CreatesImplicitConnections() || targetNodeExecutionSlot->CreatesImplicitConnections())
+        {
+            // If a connection exists between the provided endpoints, try to create an implicit connection
+            AZ::Entity* foundEntity = nullptr;
+            if (FindConnection(foundEntity, sourceEndpoint, targetEndpoint))
+            {
+                ConnectByEndpoint(sourceNodeExecutionSlot->GetEndpoint(), targetNodeExecutionSlot->GetEndpoint());
+            }
+            // If a connection doesn't exist between the provided endpoints, check if the implicit connection should be removed
+            else
+            {
+                AZStd::vector<const ScriptCanvas::Slot*> sourceNodeDataSlots = sourceNode->GetAllSlotsByDescriptor(ScriptCanvas::SlotDescriptors::DataOut(), false);
+
+                int numDataConnectionsBetween = 0;
+
+                // Count the number of data connections between the nodes of the two endpoints
+                for (const ScriptCanvas::Slot* dataSlot : sourceNodeDataSlots)
+                {
+                    for (ScriptCanvas::EndpointResolved connectedNode : sourceNode->GetConnectedNodes(*dataSlot))
```

Comment:
I recommend using a const reference here `const ScriptCanvas::EndpointResolved&` to avoid making a copy, it's a small performance improvement, but it's a good general practice